agent/bench.py

```diff
@@ -316,6 +316,35 @@ def fetch_monitor_data(self):
             traceback.print_exc()
         return lines
 
+    def _parse_pids(self, lines):
+        pids = []
+        lines = lines.strip().split("\n")
+
+        for line in lines:
+            parts = line.strip().split()
+            name, pid = parts[0], parts[1]
+            pids.append((name, pid))
+
+        return pids
+
+    def get_worker_pids(self):
+        """Get all the processes running gunicorn for now"""
+        return self._parse_pids(self.execute("ps aux | grep gunicorn")["output"])
```

Comment:
ig You are trying to find `gunicorn` processes inside container .
If that's the case, Use `docker_execute` function to run the `ps aux ....`  inside container

Reply:
As of now I was just testing py-spy outside the container and getting a dump, will look into running py-spy in a [docker container](https://github.com/benfred/py-spy/blob/master/README.md#how-do-i-run-py-spy-in-docker) as well.

Reply:
It's fine to run `py-spy` outside.

I was just talking about getting the pids of the gunicorn process for a particular bench.
The current code will grep gunicorn process of other benches as well. 

So, either you need to run `ps aux ...` inside container or use [`docker top`](https://docs.docker.com/reference/cli/docker/container/top/) to get the pids of process inside container